src/ripple/app/ledger/Ledger.cpp

```diff
@@ -1048,25 +1048,22 @@ bool pendSaveValidated (
         return true;
     }
 
-    if (isSynchronous)
-        return saveValidatedLedger(app, ledger, isCurrent);
-
-    auto job = [ledger, &app, isCurrent] (Job&) {
-        saveValidatedLedger(app, ledger, isCurrent);
-    };
-
-    if (isCurrent)
+    JobType const jobType {isCurrent ? jtPUBLEDGER : jtPUBOLDLEDGER};
+    char const* const jobName {
+        isCurrent ? "Ledger::pendSave" : "Ledger::pendOldSave"};
+
+    // See if we can use the JobQueue.
+    if (!isSynchronous &&
+        app.getJobQueue().addCountedJob (jobType, jobName,
+        [&app, ledger, isCurrent] (Job&) {
+            saveValidatedLedger(app, ledger, isCurrent);
+        }))
     {
-        app.getJobQueue().addJob(
-            jtPUBLEDGER, "Ledger::pendSave", job);
-    }
-    else
-    {
-        app.getJobQueue ().addJob(
-            jtPUBOLDLEDGER, "Ledger::pendOldSave", job);
+        return true;
     }
 
-    return true;
+    // The JobQueue won't do the Job.  Do the save synchronously.
+    return saveValidatedLedger(app, ledger, isCurrent);
```

Comment:
`JobType` and `jobName` are unnecessarily created if `isSynchronous`.

Alternate suggestion:
```
    // Do the save synchronously.
    if (isSynchronous)
        return saveValidatedLedger(app, ledger, isCurrent);

    JobType const jobType {isCurrent ? jtPUBLEDGER : jtPUBOLDLEDGER};
    char const* const jobName {
        isCurrent ? "Ledger::pendSave" : "Ledger::pendOldSave"};
    return app.getJobQueue().addJob (jobType, jobName,
        [&app, ledger, isCurrent] (Job&) {
            saveValidatedLedger(app, ledger, isCurrent);
        });
```